### PR TITLE
Add back CMake build config for macOS

### PR DESCRIPTION
**Pull request by louwers**
This builds on #2715 with some more fixes to the CMake config for macOS.

Reinstating these CMake build files is the easiest way to get Node.js working with Metal (and probably Qt too).

### CODE REVIEW

File: .github/workflows/node-ci.yml [hunk 1]
```diff
@@ -78,12 +78,12 @@ jobs:
...
             arch: arm64
-          # Disabled until Metal backend is complete
-          # A release for macOS can be made from the opengl-2 branch
           # - runs-on: macos-12
           #   arch: x86_64
-          # - runs-on: macos-13-xlarge
+          # - runs-on: macos-13-large
           #   arch: arm64
+          - runs-on: macos-14
+            arch: arm64
```
Comment (acalcutt) [inline .github/workflows/node-ci.yml:86]: Since this works can we also try 
```
          - runs-on: macos-14-large
            arch: x86_64
```

Comment (louwers) [reply .github/workflows/node-ci.yml:86]: Oh neat. Yes let's try it.

**Comment by acalcutt** [comment]
Seems like it compiles on x86_64, but the tests get this error

```
-[MTLTextureDescriptorInternal validateWithDevice:]:1357: failed assertion `Texture Descriptor Validation
MTLTextureDescriptor: Depth, Stencil, DepthStencil textures cannot be allocated with MTLStorageModeShared or MTLStorageModeManaged on this device.
```

Do you think this will only work with arm architecture?

**Comment by louwers** [comment]
@acalcutt Should work, but GitHub might run some kind of emulation and not have a Metal enabled GPU.

**Comment by louwers** [comment]
pre-commit.ci run

**Review by acalcutt — APPROVED** [review]

**Pull request merged**